prowler/providers/aws/services/ec2/ec2_service.py

```diff
@@ -496,7 +496,16 @@ def __get_launch_template_versions__(self, launch_template):
                     launch_template.versions.append(
                         LaunchTemplateVersion(
                             version_number=template_version["VersionNumber"],
-                            template_data=template_version["LaunchTemplateData"],
+                            template_data=TemplateData(
+                                user_data=template_version["LaunchTemplateData"].get(
+                                    "UserData", ""
+                                ),
+                                associate_public_ip_address=template_version[
+                                    "LaunchTemplateData"
+                                ]
+                                .get("NetworkInterfaces")[0]
```

Comment:
To avoid an error with `None[0]`, please add this:
```suggestion
                                .get("NetworkInterfaces",[])[0]
```